test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  judgment, Chapel trusts the programmer to use the loop form they
+  want.  Moreover, for some parallel computations, race conditions can
+  be benign or acceptable, so Chapel does not want to prevent a user
+  from writing such computations.
+
+  An important thing to remember about Chapel's forall-loops is that
+  they are essentially invocations of parallel iterators.  In
+  practice, those parallel iterators are themselves often written in
+  terms of the data-parallel loop forms above, or the more explicit
+  task-parallel coforall-loops that we'll cover next.
+
+
+  Task-Parallel Loops
+  ===================
+
+  Chapel has a single task-parallel loop form, the ``coforall`` loop:
+  
+  coforall loops
+  --------------
+
+  In most respects, the coforall-loop is the simplest parallel loop
+  form to explain in Chapel.  It literally creates a distinct Chapel
+  task for each iteration of the loop and then waits until those tasks
+  have each completed executing their copy of the loop body before
+  proceeding.  Mnemonically, ``coforall`` can be thought of as a
+  "concurrent forall-loop".  For example, the following coforall loop
+  will create four tasks, one for each iteration of the loop:
+
+*/
+
+  config const numTasks = 4;
+  var total: atomic int;
+  
+  coforall tid in 1..numTasks do
+    total.add(tid);
+
+  writeln("The total of the integers 1..", numTasks, " is ", total.read());
+
+/*
+
+  Because each iteration is executed by a separate task,
+  coforall-loops can synchronize between distinct iterations of the
+  loop, unlike the data-parallel loop forms.  This makes
+  coforall-loops useful as a means of creating an arbitrary number of
+  tasks that are independently doing the same, or similar, things.  In
+  practice, the parallel iterators used to define ``forall`` loops are
+  often implemented in terms of coforall-loops that create a task per
+  processor core and/or locale.  As an example, the following
+  coforall-loop creates a task per processor core across all locales:
+
+*/
+
+
+  coforall loc in Locales {
+    on loc {
+      coforall tid in 1..numTasks {
+        /* This loop body will be executed once per core per locale */
+      }
+    }
+  }
+
+/*
+
+  For further information about tasks in Chapel, see the :ref:`Task
+  Parallelism Primer <primers-taskParallel>`.
+
+  When using coforall loops, keep in mind that a task will be created
+  for each iteration, and that each task will consume memory and
+  processing resources.  For example, a coforall loop with a million
+  iterations will literally create a million tasks.  If you don't have
+  a million cores to run them on, this is likely to be overkill,
+  requiring more memory and processing power than may be warranted.
+  If there is no explicit synchronization between the iterations,
+  perhaps a forall loop would be a better choice, since it would use a
+  number of tasks proportional to the targeted hardware parallelism?
```

Comment:
This doesn't seem like a question to me, I'd probably use a period (it is a suggestion rather than a question).